packages/kbn-es-query/src/es_query/from_filters.ts

```diff
@@ -64,27 +64,32 @@ export interface EsQueryFiltersConfig {
  * @public
  */
 export const buildQueryFromFilters = (
-  filters: Filter[] = [],
-  indexPattern: DataViewBase | undefined,
+  inputFilters: Filter[] = [],
+  inputDataViews: DataViewBase | DataViewBase[] | undefined,
```

Comment:
How is this function used? Will it receive a couple of data views, or will it receive all the data views available in Kibana? if it is the former, then I think worth optimizing the function and not go through the list for each filter twice. (e.g., converting to a key map by index first)

Reply:
also if all the data views are passed, then I'd do an early return if an index is `undefined` in `findIndexPattern`

Reply:
it should not receive all the dataviews, but just some of them. for example in lens or maps each layer of data can use a separate dataview.